src/node_file.h

```diff
@@ -210,10 +287,14 @@ class FSReqPromise : public FSReqBase {
   SET_MEMORY_INFO_NAME(FSReqPromise)
   SET_SELF_SIZE(FSReqPromise)
 
+  FSReqPromise(const FSReqPromise&) = delete;
+  FSReqPromise& operator=(const FSReqPromise&) = delete;
+  FSReqPromise(const FSReqPromise&&) = delete;
+  FSReqPromise& operator=(const FSReqPromise&&) = delete;
+
  private:
   bool finished_ = false;
   AliasedBuffer<NativeT, V8T> stats_field_array_;
-  DISALLOW_COPY_AND_ASSIGN(FSReqPromise);
```

Comment:
1. It's a MACRO
2. It in `private`
3. Does not delete the `move`s

Reply:
I think it's fine to use a macro if it reduces repetition..about deleting move constructor&operator, why not just add another macro? (oops, I know)

Reply:
The move variants are implicitly deleted, according to https://github.com/nodejs/node/pull/23092

Reply:
In this case I'm in favor being explicit over less-code-via-macros.

As for the move semantics, they are not generated, but are left undefined so it's a clang-tidy error, since the compiler can't say what we ment (not implement or forgot to implement).

#23092 is sort of Ok, since the MACRO is named `DISALLOW_COPY_AND_ASSIGN` not `DISALLOW_COPY_AND_ASSIGN_AND_MOVE`

P.S. C++20 might solve this with [metaclasses](https://blog.jetbrains.com/clion/2017/11/towards-a-more-powerful-and-simpler-cpp-with-herb-sutter/)